frontend/components/SetupPage/Create/SetupCreateSafe.tsx

```diff
@@ -59,69 +53,49 @@ const CreationError = () => (
 
 export const SetupCreateSafe = () => {
   const { goto } = usePageState();
-  const { masterSafes, refetch: updateWallets } = useMasterWalletContext();
-  // const { updateMasterSafeOwners } = useMultisig();
+
+  const { selectedAgentType } = useServices();
+  const serviceTemplate = SERVICE_TEMPLATES.find(
+    (template) => template.agentType === selectedAgentType,
+  );
+
+  const {
+    masterSafes,
+    refetch: updateWallets,
+    isFetched: isWalletsFetched,
+  } = useMasterWalletContext();
   const { backupSigner } = useSetup();
 
   const masterSafeAddress = useMemo(() => {
     if (!masterSafes) return;
-    return masterSafes[0]?.address;
-  }, [masterSafes]);
+    return masterSafes.find(
+      (safe) => safe.evmChainId === asEvmChainId(serviceTemplate?.home_chain),
+    );
+  }, [masterSafes, serviceTemplate?.home_chain]);
 
   const [isCreatingSafe, setIsCreatingSafe] = useState(false);
 
-  const [gnosisFailed, setGnosisFailed] = useState(false);
```

Comment:
had to remove this b/c this is confusing, suggest rethinking this approach once optimus is planned again